Formula/terraform-ls.rb

```diff
@@ -0,0 +1,25 @@
+class TerraformLs < Formula
+  desc "Terraform Language Server"
+  homepage "https://github.com/hashicorp/terraform-ls"
+  url "https://github.com/hashicorp/terraform-ls/archive/v0.4.0.tar.gz"
+  sha256 "b0279de3016a19d6e3055c71d482a6e22856aa8cf2a70153bb8d5c335edf88ce"
+  head "https://github.com/hashicorp/terraform-ls.git"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args, "-ldflags", "-s -w"
+  end
+
+  test do
+    require "Open3"
+
+    begin
+      _, _, stderr, wait_thr = Open3.popen3("#{bin}/terraform-ls", "serve")
+      pid = wait_thr.pid
+      assert_match "Starting terraform-ls", stderr.gets("\n")
+    ensure
+      Process.kill "SIGKILL", pid
+    end
```

Comment:
This can be written as:
```suggestion
    assert_match "Starting terraform-ls", shell_output("#{bin}/terraform-ls serve")
```

However, since this is a server maybe you can make some request to it? Something like httpd?

Reply:
Making the above change results in an indefinite hang, as `terraform-ls serve` starts a daemon which does not exit on its own.

Reply:
Brew will kill any remaining processes at the end of the test.

Reply:
Perhaps it should, but in this case it does not. As I said, making that change resulted in it hanging indefinitely in my terminal window when running `brew test terraform-ls`. For what it's worth, the above code was taken from another formula within the repository that tests a daemon process.

Reply:
Does it start a deamon in the background? Or does this command start a long running process? Because in case of the latter you should do that in a `fork do` block. You can then query the server as a test.

Reply:
It starts in the foreground as the test is currently being called. I'll take a look through the repo for some other formulas which are using this pattern for testing and get it updated. Thanks.